### PR TITLE
Document channel message value conversions

### PR DESCRIPTION
**Pull request by mravn-google**
Partial fix of https://github.com/flutter/flutter/issues/8973.
Companion flutter/engine PR: https://github.com/flutter/engine/pull/3633

### CODE REVIEW

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 1]
```diff
@@ -53,6 +59,14 @@ class StringCodec implements MessageCodec<String> {
 /// * [String]s
 /// * [List]s of supported values
 /// * [Map]s from strings to supported values
+///
+/// On Android, messages are decoded using the `org.json` library.
+/// On iOS, messages are decoded using the `NSJSONSerialization` library.
+/// In both cases, the use of top-level simple messages (`null`, [bool], [num],
```
Comment (Hixie) [inline packages/flutter/lib/src/services/message_codecs.dart]: `null` shouldn't have backticks

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 2]
```diff
@@ -53,6 +59,14 @@ class StringCodec implements MessageCodec<String> {
...
+/// and [String]) is supported (by the Flutter SDK). The decoded value will be
+/// `null`/`nil` for `null`, and identical to what would result from decoding a
```
Comment (Hixie) [inline packages/flutter/lib/src/services/message_codecs.dart]: here too, at least for the Dart one.

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 3]
```diff
@@ -163,6 +177,38 @@ class JSONMethodCodec implements MethodCodec {
 /// * [Uint8List]s, [Int32List]s, [Int64List]s, [Float64List]s
 /// * [List]s of supported values
 /// * [Map]s from supported values to supported values
+///
+/// On Android, messages are represented as follows:
+///
+/// * `null`: `null`
```
Comment (Hixie) [inline packages/flutter/lib/src/services/message_codecs.dart]: ditto

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 4]
```diff
@@ -163,6 +177,38 @@ class JSONMethodCodec implements MethodCodec {
...
+/// * [Uint8List]: `byte[]`
+/// * [Int32List]: `int[]`
+/// * [Int64List]: `long[]`
+/// * [Float64List]: `double[]`
+/// * [List]: `java.util.ArrayList`
+/// * [Map]: `java.util.HashMap`
+///
+/// On iOS, messages are represented as follows:
+///
+/// * `null`: `nil`
```
Comment (Hixie) [inline packages/flutter/lib/src/services/message_codecs.dart]: ditto

**Comment by Hixie** [comment]
LGTM

yay docs!

**Comment by mravn-google** [comment]
Removed backticks from null in sibling files as well.

**Pull request merged**